src/main/java/dan200/computercraft/shared/util/InventoryUtil.java

```diff
@@ -169,7 +169,7 @@ private static ItemStack storeItems( ItemStack stack, IItemHandler inventory, in
             if( remainder == null ) break;
             remainder = inventory.insertItem( slot, remainder, false );
         }
-        return remainder;
+        return areItemsEqual( stack, remainder ) ? stack : remainder;
```

Comment:
Hrrmrm, that does violate the contract of `insertMethod` (the docs say not to modify the stack), but better safe than sorry I guess :).